src/service/referral.service.ts

```diff
@@ -55,6 +55,23 @@ class ReferralService {
 
     async createReferral(createReferralDto: CreateReferralDto): Promise<Referral> {
         const { name, email, experience, address, roleId, phone, openingId, referredById, resume } = createReferralDto;
+
+        
+        const referrals = await this.referralRepository.findReferralsByEmail(email)
+        if (referrals){
+            const referralWithSameRole = referrals.find((referral)=>referral.role.id == roleId)
+            if (referralWithSameRole){
```

Comment:
Did you formatting? (CTRL + SHIFT + I)

Reply:
No. I'll do that and push the code again